shared/AppInsightsCommon/src/Offline.ts

```diff
@@ -58,26 +93,58 @@ export function createOfflineListener(parentEvtNamespace?: string | string[]): I
                 eventOn(target, "offline", _setOffline, _evtNamespace);
             }
         }
-
         return enabled;
     }
 
+    function _isOnline(){
+        return _onlineStatus;
+    }
+
+    function currentState(){
+        if (uState === eOfflineValue.Offline || rState === eOfflineValue.Offline){
+            return false;
+        }
+        return true; // if both unknown, then we assume the network is good
+    }
+
+    function listnerNoticeCheck(){
+        // we were offline and are now online or we were online and now offline
+        if (_isOnline() !== currentState()) {
```

Comment:
Rather than call the `_isOnline()` method just reference the variable (will be slightly faster) -- but won't change operation.

maybe refactor to
```ts
    let newState = currentState();
    if (_onlineState !== newState) {
       _onlineState = newState;
...
```